chef/cookbooks/crowbar/recipes/crowbar-db-dump.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Cookbook Name:: crowbar
+# Recipe:: crowbar-db-dump
+#
+# Copyright 2013-2016, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This recipe is for executing actions that need to be done at nodes
+# as a preparation for the system upgrade. More specifically, this is executed
+# for nodes that have database installed because they need to be handled differently
+# than non DB nodes.
+#
+# First the database content should be dumped. In the next run (indicated by extra argument),
+# the database needs to be shut down.
+
+return unless node[:platform] == "suse"
+
+if node["crowbar_wall"]["crowbar_openstack_shutdown"]
+
+  # 1. Stop all BUT postgresql pacemaker resources in this cluster
+  bash "stop pacemaker resources" do
+    code <<-EOF
+      for type in clone ms primitive; do
+        for resource in $(crm configure show | grep ^$type | grep -Ev "postgresql|vip-admin-database" | cut -d " " -f2); do
+          crm resource stop $resource
+        done
+      done
+    EOF
+    only_if { ::File.exist?("/usr/sbin/crm") }
+  end
+
+  # 2. Stop openstack services on this node
+  # Note that for HA setup, they should be already stopped by pacemaker
+  # Services running on other nodes should be already stopped by previous run of crowbar-upgrade
+  bash "stop HA and openstack services" do
+    code <<-EOF
+      for i in /etc/init.d/openstack-* \
+               /etc/init.d/openvswitch-switch \
+               /etc/init.d/ovs-usurp-config-* \
+      do
+        test -e $i && $i stop
+      done
+    EOF
+  end
+
+  # If postgres is not running here, it means we're in the DB cluster and database runs
+  # on the other node: let the other node take care of the rest.
+  return unless ::Kernel.system("service postgresql status")
+
+  # 3. Dump the database at the DB node
+  # FIXME check the available space before: sudo -u postgres pg_dumpall | wc -c
+  # FIXME find the correct place for the dump (drbd?)
+
+  dump_location = "/var/lib/pgsql/db-before-cloud6.dump"
+
+  # if we have drbd, find out the device
+  # FIXME what if we have shared NFS?
+  if ::Kernel.system("service drbd status")
```

Comment:
Why is this needed at all? If I understand correctly, we are already sure that postgresql is running here. And AFAIK we always store the data in the same location (/var/lib/pgsql). Can't we just assume that if postgresql is running here the correct device is mounted there?